src/main/java/com/google/sps/servlets/RetrieveActivitiesServlet.java

```diff
@@ -53,42 +55,17 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   /**
   * Retrieves type of activity from request parameters.
   */
-  private static String getActivityType(HttpServletRequest request) {
+  private static Category getActivityCategory(HttpServletRequest request) {
     String name = "activity-type", defaultValue = "";
-    return getParameter(request, name, defaultValue);
-  }
-
-  private static List<Activity> getGames() {
-    List<Activity> games = Arrays.asList(
-      new Activity("Pictionary", GAMES_CATEGORY, "www.pictionary.com"),
-      new Activity("Spy Master", GAMES_CATEGORY, "www.spymaster.com"),
-      new Activity("Skribbl", GAMES_CATEGORY, "www.skribbl.io")
-    );
-
-    return games;
-  }
-
-  private static List<Activity> getVideos() {
-    List<Activity> videos = Arrays.asList(
-      new Activity("Meditation for Anxiety", VIDEOS_CATEGORY, "https://www.youtube.com/watch?v=4pLUleLdwY4"),
-      new Activity("Restorative Yoga and Meditation", VIDEOS_CATEGORY, "https://www.youtube.com/watch?v=LI6RwT0ulDk"),
-      new Activity("Zumba Workout", VIDEOS_CATEGORY, "https://www.youtube.com/watch?v=-VXhoeaxxi0")
-    );
+    String activityType = getParameter(request, name, defaultValue);
+    Category activityCategory = null;
 
-    return videos;
-  }
-
-  private static List<Activity> getArticles() {
-    List<Activity> articles = Arrays.asList(
-      new Activity("How to Improve Your Psychological Well-Being", ARTICLES_CATEGORY, 
-        "https://www.verywellmind.com/improve-psychological-well-being-4177330"),
-      new Activity("How to Know if Zen Meditation Is Right for You", ARTICLES_CATEGORY, 
-        "https://www.verywellmind.com/what-is-zen-meditation-4586721"),
-      new Activity("What Is the Negativity Bias?", ARTICLES_CATEGORY, 
-        "https://www.verywellmind.com/negative-bias-4589618")
-    );
-
-    return articles;
+    try {
+      activityCategory = Category.valueOf(activityType.toUpperCase());
+    } catch(IllegalArgumentException exception) {
+      System.out.println("The specified activity type was either not defined or it's invalid.");
```

Comment:
It will be more helpful to include the activity type string in this error message, instead of saying "the specified activity type".

Reply:
Fixed it.